crypto/ecdsa/signer.go

```diff
@@ -0,0 +1,86 @@
+package neofsecdsa
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/sha512"
+	"fmt"
+
+	"github.com/nspcc-dev/neo-go/pkg/crypto/keys"
+	neofscrypto "github.com/nspcc-dev/neofs-sdk-go/crypto"
+)
+
+// Signer implements neofscrypto.Signer based on ECDSA.
+//
+// Supported schemes:
+//  - neofscrypto.ECDSA_SHA512 (default)
+//  - neofscrypto.ECDSA_DETERMINISTIC_SHA256
+//
+// Instances MUST be initialized with ecdsa.PrivateKey using SetKey.
+type Signer struct {
+	deterministic bool
+
+	key ecdsa.PrivateKey
+}
+
+// SetKey specifies ecdsa.PrivateKey to be used for ECDSA signature calculation.
+func (x *Signer) SetKey(key ecdsa.PrivateKey) {
+	x.key = key
+}
+
+// MakeDeterministic makes Signer to use Deterministic ECDSA scheme
+// (see neofscrypto.ECDSA_DETERMINISTIC_SHA256). By default,
+// neofscrypto.ECDSA_SHA512 is used.
+func (x *Signer) MakeDeterministic() {
```

Comment:
This method looks very strange, given that `deterministic` is not a concise description of an algorithm we use. Let's stick with different concrete types for different algorithms  (could be aliases to `ecdsa.PrivateKey` then).

Reply:
We can provide separate `neofsecdsa.SignerRFC6979` type. But IMO proposed method is pretty understandable.

Thoughts @alexvanin @carpawell

Reply:
I mean, those are different signature algorithms, with different hash digest types. If I were a client I am not sure I would understand what `MakeDeterministic` means. `SignerRFC6979` is much more descriptive.

Reply:
did not think about the confusion when looked at it but now agree that such a naming may lead to misunderstanding

Reply:
Refactored to per-scheme types.